components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/store/connector/RDBMSConnector.java

```diff
@@ -0,0 +1,194 @@
+import org.wso2.micro.integrator.initializer.handler.transaction.exception.TransactionCounterInitializationException;
+import org.wso2.micro.integrator.initializer.handler.transaction.security.CryptoUtil;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Date;
+import javax.crypto.Cipher;
+import javax.sql.DataSource;
+
+/**
+ * The connector class which deals with underlying internal transaction related tables.
+ */
+public class RDBMSConnector {
+
+    private static final Log LOG = LogFactory.getLog(RDBMSConnector.class);
+    private DataSource dataSource;
+    private String nodeId;
+    private Cipher cipher;
+
+    /**
+     * Constructor.
+     *
+     * @param dataSource - The datasource config to initiate the connection.
+     * @throws TransactionCounterInitializationException - when something goes wrong while initializing RDBMS
+     *                                                   connection.
+     */
+    public RDBMSConnector(DataSource dataSource, String nodeId, Cipher cipher)
+            throws TransactionCounterInitializationException {
+        this.nodeId = nodeId;
+        this.cipher = cipher;
+        this.dataSource = dataSource;
+        try (Connection connection = getConnection()) {
+            DatabaseMetaData metaData = connection.getMetaData();
+            String databaseType = metaData.getDatabaseProductName();
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Successfully connected to : " + databaseType);
+            }
+        } catch (SQLException ex) {
+            throw new TransactionCounterInitializationException("Error while initializing RDBMS connection.", ex);
+        }
+    }
+
+    /**
+     * Get connection.
+     *
+     * @return - Connection with auto commit true.
+     * @throws SQLException -
+     */
+    private Connection getConnection() throws SQLException {
+
+        Connection connection = dataSource.getConnection();
+        connection.setAutoCommit(true);
+        return connection;
+    }
+
+    /**
+     * Add transaction count to the database.
+     */
+    public void addTransaction() throws SQLException {
+        // if raw exists - update else and a new raw.
+        if (checkDataExists()) {
+            updateStats();
+        } else {
+            addNewRow();
+        }
+    }
+
+    /**
+     * Get transaction count of a given year and month.
+     *
+     * @param year        Year.
+     * @param monthNumber Month.
+     * @return Aggregated transaction count if exists. Else (-1)
+     * @throws SQLException Error occurred while fetching data from the database.
+     */
+    public long getTransactionCountOfMonth(int year, int monthNumber) throws SQLException {
+
+        String dateString = year + "-" + monthNumber + "-01";
```

Comment:
I hope this format is tested with all db types?

Reply:
@GDLMadushanka has tested it with all db types.